app/controllers/contacts_controller.rb

```diff
@@ -99,6 +101,52 @@ def contact
     end
   end
 
+  def edit_profile_action
+    formValues = JSON.parse(params.require(:formValues), symbolize_names: true)
+    edited_profile_details = formValues[:editedProfileDetails]
+    edit_profile_reason = formValues[:editProfileReason]
+    contact_email = formValues[:contactEmail]
+    attachment = params[:attachment]
+    wca_id = formValues[:wcaId]
+    if current_user.present?
+      profile_to_edit = {
+        name: current_user.person.name,
+        country_iso2: current_user.person.country_iso2,
+        gender: current_user.person.gender,
+        dob: current_user.person.dob,
+      }
+    else
```

Comment:
Would it not be prudent to perform some kind of sanity checking whether the currently logged in user has actually claimed the WCA ID they're trying to edit?

Right now, you're just happily loading the current user's details without any check, so I could impersonate anybody by simply entering their WCA ID